metrics/metrics.go

```diff
@@ -51,6 +53,21 @@ func (m *LabelMap) Get(key string) *expvar.Int {
 	return m.Map.Get(key).(*expvar.Int)
 }
 
+// GetShardedInt returns a direct pointer to the syncs.ShardedInt for key,
+// creating it if necessary.
+//
+// Note: this method can race, so early modifications may be lost under
+// contention, this is unavoidable until expvar provides an atomic set operation
+// for arbitrary expvar.Var values.
```

Comment:
Ugh this is annoying; I read the `expvar` code and yeah, it looks like there's no good way to do that. We could consider adding some synchronization ourselves?